spec.bs

```diff
@@ -452,6 +457,93 @@ To <dfn>build an interest group permissions url</dfn> given a [=origin=] |ownerO
 
 </div>
 
+<h3 id="interest-group-storage-maintenance">Interest Group Storage Maintenance</h3>
+
+There is a job that periodically [=performs storage maintenance=] on the [=user agent=]'s
+[=interest group set=]. It performs operations such as [=list/removing=] expired or excess
+[=interest groups=]. An [=interest group set=] must have no more than:
+  * 2000 [=interest group/owners=];
+  * 1000 [=regular interest groups=] per [=interest group/owner=];
+  * 20000 [=negative interest groups=] per [=interest group/owner=];
+  * <code>10\*1024\*1024</code> total [=interest group/estimated size=] of [=interest groups=] per
+    [=interest group/owner=].
+
+<div algorithm>
+To <dfn>perform storage maintenance</dfn>:
+
+1. Let |ownersAndExpiry| be a new [=ordered map=] whose [=map/keys=] are [=origins=] and

```

Comment:
Can we add a note above this part of the algorithm that explains what this is doing? E.g. "Construct a set of [=interest group/owner=]s whose [=interest group=]s should all be removed from the [=user agent=]'s [=interest group set=] because the number of distinct [=interest group/owner=]s in the [=user agent=]'s [=interest group set=] exceeds the limit specified above. This retains those [=interest group/owner=]s whose [=interest group=]s have an [=interest group/expiry=] furthest in the future."